File: trunk/ia/get_entry.php
<?php
// get_entry.php for get_entry in /u/all/veau-g_a/cu/rendu/c/zappy2010/ia
// 
// Made by adrien veau-greiner
// Login   <veau-g_a@epitech.net>
// 
// Started on  Thu May  6 15:18:58 2010 adrien veau-greiner
// Last update Fri May  7 11:07:16 2010 adrien veau-greiner
//

function	get_team_name($arg)
{
  if ($arg == "GRAPHIC" || $arg == "graphic")
    {
      aff_error("name \"GRAPHIC\" or \"graphic\" is reserved !!!");
      return (-1);
    }
  if ($arg == "-p" || $arg == "-n" || $arg == "-h")
    return (-1);
  return ($arg);
}

function	get_port($arg)
{
  if ($arg == "-p" || $arg == "-n" || $arg == "-h")
    return (-1);
  if ((is_numeric($arg) == FALSE) || ($argv > 65535))
    {
      aff_error("port ".$arg." is not a valid port !!!");
      return (-1);
    }
  return ($arg);
}

function	get_host($arg)
{
  if ($arg == "-p" || $arg == "-n" || $arg == "-h")
    return (-1);
  return ($arg);
}

function	get_args($argv, $port, $team, $host)
{
  $i = 0;
  while ($argv[$i])
    {
      if ($argv[$i] == "-n" && isset($argv[$i + 1]) == TRUE)
	if (($team = get_team_name($argv[$i + 1])) == -1)
	  return (FALSE);
      if ($argv[$i] == "-p" && isset($argv[$i + 1]) == TRUE)
	if (($port = get_port($argv[$i + 1])) == -1)
	  return (FALSE);
      if ($argv[$i] == "-h" && isset($argv[$i + 1]))
	if (($host = get_host($argv[$i + 1])) == -1)
	  return (FALSE);
      $i++;
    }
  return (TRUE);
}

function	get_entry($argc, $argv, $p)
{
  if (($argc < 4) || ($argc > 7))
    return (FALSE);
  if (get_args($argv, &$port, &$team, &$host) == FAlSE)
    return (FALSE);
  if ($host == NULL && $argc != 5)
    return (FALSE);
  if ((isset($port) == TRUE) && (isset($team) == TRUE))
    {
      if (isset($host) == FALSE)
	{
	  if ($p->init(1, $team, $port) == FALSE)
	    return (FALSE);
	}
      else
	if ($p->init(1, $team, $port, $host) == FALSE)
	  return (FALSE);
    }
  else
    return (FALSE);
  return (TRUE);
}

?>

File: trunk/graphic/sources/get_entry.c
/*
** get_entry.c for get_entry in /u/all/veau-g_a/cu/rendu/c/zappy2010/graphic/sources
** 
** Made by adrien veau-greiner
** Login   <veau-g_a@epitech.net>
** 
** Started on  Thu May 13 12:30:39 2010 adrien veau-greiner
** Last update Thu May 13 16:58:11 2010 adrien veau-greiner
*/

#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <unistd.h>
#include <netdb.h>
#include <errno.h>

#include "client.h"

int	get_port(char *p, int *cl_p)
{
  int	num_p;

  num_p = atoi(p);
  if (num_p > MAX_PORT || num_p <= 0)
    {
      printf("Error: invalid port %s\n", p);
      return (-1);
    }
  *cl_p = num_p;
  return (0);
}

int	get_host(char *h, t_client *cl)
{
  if ((cl->host = gethostbyname(h)) == NULL)
    {
      printf("\033[31mError => %s\n\033[00m", hstrerror(h_errno));
      return (-1);
    }
  return (0);
}

int	get_entry(char **av, t_client *cl)
{
  int	i;
  int	flag;

  i = 0;
  flag = 0;
  while (i < 4)
    {
      if ((strcmp(av[i], "-p") == 0) && av[i + 1])
	{
	  if (get_port(av[i + 1], &cl->port) == -1)
	    return (-1);
	  flag += 1;
	}
      if ((strcmp(av[i], "-h") == 0) && av[i + 1])
	{
	  if (get_host(av[i + 1], cl) == -1)
	    return (-1);
	  flag += 2;
	}
      i++;
    }
  if (flag != 3)
    return (aff_help());
  return (0);
}
